hooks/useContract.ts

```diff
@@ -0,0 +1,24 @@
+import { useState, useEffect } from 'react';
+import { Address, PublicClient, getContract } from 'viem';
+
+export function useContract<TAbi extends readonly unknown[]>(address: Address, abi: TAbi, publicClient: PublicClient) {
+  const [contract, setContract] = useState(() =>
+    getContract({
+      address,
+      abi,
+      client: publicClient,
+    })
+  );
+
+  useEffect(() => {
+    setContract(
+      getContract({
```

Comment:
It's usually good practice to have effects as named functions

```suggestion
  useEffect(function saveContractInState() {
    setContract(
      getContract({
```

We could even enforce this via ESLint rule.

Reply:
agree